fip-0017.md

```diff
@@ -0,0 +1,265 @@
+---
+fip: 17
+title: FIO Token and Domain Wrapping
+status: Draft
+type: Functionality
+author: Pawel Mastalerz <pawel@dapix.io>
+created: 2020-09-17
+updated: 2020-09-17
+---
+
+# Abstract
+Functionality proposed in this FIP is intended to make wrapping of FIO Tokens and Domains easier.
+
+Proposed new actions:
+|Action|Endpoint|Description|
+|---|---|---|
+|wrapdomain|wrap_fio_domain|Transfers FIO Domain to designated FIO account.|
+|wraptoken|wrap_fio_token|Transfers FIO Token to designated FIO account.|
+||get_wrapped_fio_domains|Returns FIO Domains wrapped to account.|
+||get_wrapped_fio_tokens|Returns FIO Tokens wrapped to account.|
+
+# Motivation
+It is believed that the ability to wrap FIO Tokens and Domain NFTs will be beneficial to the FIO Community. It will open up new use cases for FIO Tokens and Domains such as:
+* Enable FIO Domains to be traded on NFT trading sites such as Open Sea as ERC-721s.
+* Enable FIO tokens to be used in a rapidly growing Defi ecosystem such as Uniswap which rewards token holders for providing liquidity to decentralized exchanges.
+
+Although wrapping cannot be accomplished entirely inside the FIO Protocol and therefore will not be possible with this FIP alone, it is intended for this FIP to make it easier. For detailed overview of the greater initiative see [Enable FIO Token and Domain NFT wrapping](https://fioprotocol.atlassian.net/wiki/spaces/WP/pages/7012357/Enable+FIO+Token+and+Domain+NFT+wrapping).
+
+The functionality proposed in this FIP will allow anyone to deploy wrapped FIO Tokens and Domains on any chain, as long as they build the other components outside of the FIO Protocol such as other blockchain wrapping smart contract, oracle functionality.
+
+# Specification
+## New actions
+### Wrap FIO Domain
+Transfers FIO Domain to designated FIO account.
```

Comment:
Is this the right description as a wrap isn't really transferring to another FIO account but to an external chain account, right?